lib/hotel.rb

```diff
@@ -0,0 +1,101 @@
+require_relative "rooms"
+require 'date'
+require 'pry'
+
+class Hotel
+  attr_reader :rooms, :reservations
+  def initialize
+    @reservations = []
+    @rooms = []
+    (1..20).each do |room|
+      @rooms.push(Room.new(room, 200))
+    end
+  end
+
+
+  def make_reservation(check_in, check_out, rooms)
+    unavailable_rooms = []
+
+    if room_flagged_available?(check_in, check_out, rooms, unavailable_rooms)
+      @reservations.push(Reservation.new(check_in, check_out, rooms))
+    else
+      begin
```

Comment:
Refer back to our [exception handling lecture](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/02-intermediate-ruby/exception-handling.md) about how to set up this exception handling